### PR TITLE
Fix IDs when creating IconDirectoryResource from an .ico file

### LINKED ISSUE #48: IconDirectoryResource.SaveTo has bugs with some .ico file.
```
   public static void ReplaceIcon(string exeFile, string icoFile)
    {
        ResourceInfo ri = new ResourceInfo();
        ri.Load(exeFile);
        ri.Unload();
        var res=ri[Kernel32.ResourceTypes.RT_GROUP_ICON][0];
        IconDirectoryResource idr = new IconDirectoryResource(new IconFile(icoFile));
        idr.Name = res.Name;
        idr.Language = res.Language;
        idr.SaveTo(exeFile);
    }
```

attach file 0.zip contains a 0.ico
If ReplaceIcon with 0.ico, the target exe will lose icon.
But if I compile a exe with 0.ico by visual studio,it's ok ,so 0.ico is format well.
So, bug?

[0.zip](https://github.com/dblock/resourcelib/files/439457/0.zip)

### PR DESCRIPTION
**Pull request by jairbubbles**
Note: I had to tweak the solution to make it compile with VS2017. 

Fixes #48

### CODE REVIEW

**Comment by dblock** [comment]
Great. Care to update https://github.com/resourcelib/resourcelib/blob/master/CHANGELOG.md please?

Also I haven't been actively writing Windows code in a LONG time, I'd love some help with releasing the next version and such. Want in @jairbubbles?

**Comment by jairbubbles** [comment]
Ahah I see the trap! :smile: But sure I can help a little.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+### 1.7
```
Comment (dblock) [inline CHANGELOG.md]: Add `(Next)` next to the number,.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
...
+
+**Bugs**
+
+* [#48](https://github.com/resourcelib/resourcelib/pull/56) Fixed IconDirectoryResource.SaveTo has bugs with some .ico file
```
Comment (dblock) [inline CHANGELOG.md]: Make this the same format as other lines please, with your name, dashes and colons, please.

**Comment by jairbubbles** [comment]
Could not find any doc confirming this but maybe having a resource with an Id equals to 0 is what leads to issues in explorer.exe.

(https://blogs.msdn.microsoft.com/oldnewthing/20120720-00/?p=7083)

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
+### 1.7 (Next)
+
+**Bugs**
+
+* [#48](https://github.com/resourcelib/resourcelib/pull/56): Fixed `IconDirectoryResource.SaveTo` has bugs with some .ico file - [@jairbubbles](https://github.com/jairbubbles).
```
Comment (dblock) [inline CHANGELOG.md]: The number is wrong, it says 48, but PR is 56 in the link. I know I know 🙄

Comment (jairbubbles) [reply CHANGELOG.md]: Eagle eye! I just indicated the issue number as it was a #. It was not consistent in the file so I mixed both. I'll put only issue number.

**Comment by dblock** [comment]
@jairbubbles Added you as collaborator, https://github.com/resourcelib/resourcelib/invitations

 @icnocop @thoemmi We need a RELEASING doc ala https://github.com/dblock/ruby-enum/blob/master/RELEASING.md for someone new to make a release, care to help?

**Comment by thoemmi** [comment]
Sounds reasonable.

Additionally I would like to introduce some more automation:
* use [GitVersion](https://github.com/GitTools/GitVersion) to generate semantic versions
* allow AppVeyor to publish NuGet packages automatically (release packages from `master` branch and pre-releases from `develop` branch)

What do you think?

**Comment by jairbubbles** [comment]
@thoemmi "AppVeyor + GitVersion" definitely sounds like a good combo!

**Comment by thoemmi** [comment]
Yeah, we could even get rid of that weird `Version` project...

**Comment by icnocop** [comment]
@thoemmi, I'm wondering if we can publish pre-release NuGet packages from the `master` branch without actually requiring a `develop` branch? I'm just basically wondering if we can reduce the complexity involved in having multiple branches. In which scenario(s) would having both branches be helpful?

**Comment by jairbubbles** [comment]
We can also rely on git tagging on master. It's not a big project, I don't think a `dev` branch is necessary neither.

**Pull request merged**